src/components/popovers/SwapSettingsPopover.vue

```diff
@@ -1,3 +1,47 @@
+<script lang="ts">
+export enum SwapSettingsContext {
```

Comment:
may be there is some better place for this, `swap.service.ts`?

Reply:
`SwapSettingsContext` is the type of a prop in `SwapSeetingsPopover` component so I think that it makes sense to leave it there.
 
Look how you only import SwapSettingsContext when using the SwapSeetingsPopover component (high cohesion). If some other code needed this type outside this scope, then I would extract it but, that's not the case here.

Reply:
I am just not a big fan of exporting types from .vue files, seems they shouldn't be responsible only for it. but yeah, may be I'm wrong here

Reply:
I also don't like that pattern but, in this concrete case I think it is the less worst option. 

Moving the type to a service seems off IMO

Moving it to a new file `SwapSeetingsPopover.d.ts` would be my next option. But I prefer to have the type self-contained within the component that uses it. 

@garethfuller What do you think?

Reply:
I don't have a strong opinion here. I think exporting types highly relevant to a component from the component file is ok.